### PR TITLE
add docs page on RPATH support (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
preview at http://boegel-eb.readthedocs.io/en/rpath_docs/RPATH-support.html

### CODE REVIEW

**Comment by pescobar** [comment]
@boegel  lgtm

**Comment by boegel** [comment]
Thanks for the review @pescobar!

File: docs/RPATH-support.rst [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+.. _rpath_support_why:
+
+Why RPATH?
+----------
+
+Using RPATH can be interesting for a number of reasons:
+
+* it can help to avoid a (too) large environment, since:
+
+  * ``$LD_LIBRARY_PATH`` does not need to set anymore for all dependencies providing libraries
```
Comment (verdurin) [inline docs/RPATH-support.rst]: "need to be set anymore"

File: docs/RPATH-support.rst [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+* ensure that the library subdirectories (``lib``, ``/lib64``) of the installation directory also have an RPATH entry
+* include additional  arguments related to RPATH (e.g. ``--disable-new-dtags``)
+
+As such, ``ps`` may show something like::
+
+  \_ /bin/bash /tmp/eb-M3393U/tmpRVJqwr/rpath_wrappers/gcc -O2 example.c -L/example -lexample
+  |  \_ /example/software/GCCcore/4.9.3/bin/gcc -Wl,-rpath=$ORIGIN/../lib -Wl,-rpath=$ORIGIN/../lib64 -Wl,--disable-new-dtags -Wl,-rpath=/example -O2 example.c -L/example -lexample
+
+Here, ``/tmp/eb-M3393U/tmpRVJqwr/rpath_wrappers/gcc`` is the wrapper script for ``gcc``,
+which tweakes the list of command line arguments for ``gcc``
```
Comment (verdurin) [inline docs/RPATH-support.rst]: "tweaks"

File: docs/RPATH-support.rst [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+            eb --rpath-filter='/lib.*,/usr.*,/example.*'
+
+
+.. _rpath_support_LD_LIBRARY_PATH:
+
+Relation to ``$LD_LIBRARY_PATH``
+--------------------------------
+
+As mention above (:ref:`rpath_support_why`), using RPATH avoids the need to update ``$LD_LIBRARY_PATH`` for every dependency.
```
Comment (verdurin) [inline docs/RPATH-support.rst]: "mentioned"

**Pull request merged**